src/crimson/os/seastore/segment_manager/zns.cc

```diff
@@ -56,21 +57,30 @@ static zns_sm_metadata_t make_metadata(
   
   auto config_segment_size = get_conf<Option::size_t>(
     "seastore_segment_size");
-  INFO("config segment size: {}", config_segment_size);
-  size_t zones_per_segment = config_segment_size / zone_capacity;
+
+  size_t segment_size = (zone_size == 0) ? config_segment_size : (zone_size << SECT_SHIFT);
+  size_t zones_per_segment = segment_size / (zone_size << SECT_SHIFT);
   size_t segments = (num_zones - 1) * zones_per_segment;
```

Comment:
`size_t segments = (num_zones - 1) / zones_per_segment;`

Reply:
I think it might be a useful tunable to allow a segment to contain multiple zones. @athanatos what do you think?

Reply:
There are zns implementations where you really need to be writing across 10s of zones to get full throughput -- I think a good way to do that will be to let a segment be multiple zones and stripe the incoming write stream across them.